packages/utilities/src/merge.ts

```diff
@@ -0,0 +1,30 @@
+/**
+ * Simple deep merge function. Takes all arguments are returns a new object which merges
+ * them all, in the order provided.
+ */
+export function merge<T = {}>(...args: Partial<T | null | undefined | false>[]): T {
```

Comment:
The code supports arguments that are not all of the same type T but the typings do not have this flexibility. You should do something similar to the typings for mergeStyleSets/concatStyleSets

Reply:
The BIG problem is that you cannot use `merge` without explicitly typing the shape of the fully merged object atm if some of the arguments are Partials of the final object, and it makes it really hard to use this utility function.